File: src/test/java/io/github/binout/soccer/infrastructure/persistence/InMemorySeasonRepository.java
package io.github.binout.soccer.infrastructure.persistence;

import io.github.binout.soccer.domain.season.Season;
import io.github.binout.soccer.domain.season.SeasonRepository;

import javax.enterprise.context.ApplicationScoped;
import javax.enterprise.inject.Vetoed;
import java.util.Map;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;
import java.util.stream.Stream;

@ApplicationScoped
@Vetoed
public class InMemorySeasonRepository implements SeasonRepository {

    private Map<String, Season> seasons;

    public InMemorySeasonRepository() {
        seasons = new ConcurrentHashMap<>();
    }

    @Override
    public void add(Season season) {
        seasons.put(season.name(), season);
    }

    @Override
    public Optional<Season> byName(String name) {
        return Optional.ofNullable(seasons.get(name));
    }

    @Override
    public Stream<Season> all() {
        return seasons.values().stream();
    }

}

File: src/test/kotlin/io/github/binout/soccer/infrastructure/persistence/InMemorySeasonRepository.kt
package io.github.binout.soccer.infrastructure.persistence

import io.github.binout.soccer.domain.season.Season
import io.github.binout.soccer.domain.season.SeasonRepository
import java.util.Optional
import java.util.concurrent.ConcurrentHashMap
import java.util.stream.Stream

class InMemorySeasonRepository : SeasonRepository {

    private val seasons: MutableMap<String, Season>

    init {
        seasons = ConcurrentHashMap()
    }

    override fun add(season: Season) {
        seasons[season.name] = season
    }

    override fun byName(name: String): Season? = seasons[name]

    override fun all(): List<Season> = seasons.values.toList()

}
